_rules/auto-update-text-efbfc7.md

```diff
@@ -0,0 +1,389 @@
+			clearInterval(updates)
+		}
+	</script>
+</body>
+```
+
+#### Passed Example 2
+
+The text content automatically updates every 3 seconds after the page completes loading. A button is available to pause and resume the automatic updates.
+
+```html
+<body onload="toggle()">
+	<p>Random number: <span id="target">1</span></p>
+	<input type="button" id="control" onclick="toggle()" value="Pause updates" />
+
+	<p>
+		The W3C Web Accessibility Initiative (WAI) develops standards and support materials to help you understand and
+		implement accessibility.
+	</p>
+
+	<script type="text/javascript">
+		function change() {
+			var target = document.getElementById('target')
+			var number = Math.floor(Math.random() * 1000)
+			target.innerText = number
+		}
+
+		var updates
+		var updating = false
+
+		function toggle() {
+			var control = document.getElementById('control')
+			if (updating) {
+				control.value = 'Resume updates'
+				updating = false
+				clearInterval(updates)
+			} else {
+				control.value = 'Pause updates'
+				updating = true
+				updates = setInterval(change, 3000)
+			}
+		}
+	</script>
+</body>
+```
+
+#### Passed Example 3
+
+The text content automatically updates every 3 seconds after the page completes loading. A button is available to hide the automatically updating content.
+
+```html
+<body onload="start()">
+	<p>Random number: <span id="target">1</span></p>
+	<input type="button" onclick="hide()" value="Hide updates" />
+
+	<p>
+		The W3C Web Accessibility Initiative (WAI) develops standards and support materials to help you understand and
+		implement accessibility.
+	</p>
+
+	<script type="text/javascript">
+		function change() {
+			var target = document.getElementById('target')
+			var number = Math.floor(Math.random() * 1000)
+			target.innerText = number
+		}
+
+		function start() {
+			setInterval(change, 3000)
+		}
+
+		function hide() {
+			var target = document.getElementById('target')
+			target.style.visibility = 'hidden'
+		}
+	</script>
+</body>
+```
+
+### Failed
+
+#### Failed Example 1
+
+The text content automatically updates every 3 seconds after the page completes loading. There is no component to stop or pause the automatic updates.
+
+```html
+<body onload="start()">
+	<p>Random number: <span id="target">1</span></p>
+
+	<p>
+		The W3C Web Accessibility Initiative (WAI) develops standards and support materials to help you understand and
+		implement accessibility.
+	</p>
+
+	<script type="text/javascript">
+		function change() {
+			var target = document.getElementById('target')
+			var number = Math.floor(Math.random() * 1000)
+			target.innerText = number
+		}
+
+		function start() {
+			updates = setInterval(change, 3000)
+		}
+	</script>
+</body>
+```
+
+#### Failed Example 2
+
+The text content (part of a progress bar) automatically updates every second after the page completes loading. There is no component for the user to pause the auto-updates, other than waiting for them to end.
```

Comment:
I don't think that the core code for this example should be different from the others.
The same random number stuff can be used, no need for the "pick a price" stuff, just random number is OK.
The progress bar can probably be replaced by a timer displaying the remaining time. That should remove most or all of the styling and trim down the example.
Actually, the progress bar can probably get completely removed. It is only need because you have a button for stopping the random number, but if that button is removed, then the progress bar can be remove (there will be a random number auto updating for some time with no way to stop it, which is enough to fail the rule).

Reply:
I guess that this is a legacy test from a previous version of the rule where the progress bar made sense to have in a test. Given that the rule has been simplified, we don't even need this test anymore (if I followed your suggestions - with which I agree - we would end up with a test that was equivalent to failed example 1). I'm going to remove this.